Use pkg-config for explicit system-library builds on macOS, keeping
libxml2/libxslt headers and libraries from the same provider.

--- a/configure.ac
+++ b/configure.ac
@@ -10494,7 +10494,7 @@
 if test "$with_system_libxml" = "yes"; then
     AC_MSG_RESULT([external])
     SYSTEM_LIBXSLT=TRUE
-    if test "$_os" = "Darwin"; then
+    if test "$_os" = "Darwin" -a "$with_system_libs" != "yes"; then
         dnl make sure to use SDK path
         LIBXSLT_CFLAGS="-I$MACOSX_SDK_PATH/usr/include/libxml2"
         LIBEXSLT_CFLAGS="$LIBXSLT_CFLAGS"
@@ -10541,7 +10541,7 @@
 if test "$with_system_libxml" = "yes"; then
     AC_MSG_RESULT([external])
     SYSTEM_LIBXML=TRUE
-    if test "$_os" = "Darwin"; then
+    if test "$_os" = "Darwin" -a "$with_system_libs" != "yes"; then
         dnl make sure to use SDK path
         LIBXML_CFLAGS="-I$MACOSX_SDK_PATH/usr/include/libxml2"
         dnl omit -L/usr/lib
